src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterJsonElement.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class JsonValueConverterJsonElement : JsonValueConverter<JsonElement>
+    {
+        public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out JsonElement value)
+        {
+            if (JsonDocument.TryParseValue(ref reader, out JsonDocument document))
```

Comment:
As discussed this won't work when we have a stream. It will only work with primitives where we are guaranteed the reader won't ask for more data (and potentially fail)

Reply:
As discussed, we'll handle the Stream case in a future commit